packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -789,3 +806,56 @@ export async function DocsQA<ChunkMetadata extends Jsonifiable = Jsonifiable>(pr
     </ChatCompletion>
   );
 }
+
+const ResultSchema: z.ZodObject<any> = z
+  .object({
+    answer: z.string().describe("The answer to the user's question"),
+    sources: z.array(z.string()).describe('The title or URL of each document used to answer the question'),
+  })
+  .required({ answer: true });
+
+export type QAWithSourcesResult = z.infer<typeof ResultSchema>;
+
+/** A default component for formatting DocsQAWithSources results. */
+function DefaultQAResultFormatter(result: QAWithSourcesResult) {
+  if (result.sources?.length) {
+    return `${result.answer}\n\nSources:\n${result.sources.join('\n')}`;
+  }
+  return result.answer;
+}
+
+/**
+ * Similar to {@link DocsQA}, but encourages the LLM to return sources for its answer.
+ */
+export async function* DocsQAWithSources<ChunkMetadata extends Jsonifiable = Jsonifiable>(
```

Comment:
```suggestion
export async function* DocsQAWithCitations<ChunkMetadata extends Jsonifiable = Jsonifiable>(
```

Minor, but I think this name could be clearer. All docs QA has sources, right? But the name `citations` to me is a little more evocative of citing those sources in the response. (not a blocker / do whatever you want.)